apple-silicon: panic when updating runners of a server due to wrong convert

Error log

Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-scaleway_v2.69.0 plugin:

panic: interface conversion: interface {} is []interface {}, not []string

goroutine 76 [running]:
github.com/scaleway/terraform-provider-scaleway/v2/internal/services/applesilicon.ResourceAppleSiliconServerUpdate({0x1018d0570, 0x140002dc460}, 0x1400081e090, {0x1017c7800, 0x140005588b8})
        github.com/scaleway/terraform-provider-scaleway/v2/internal/services/applesilicon/server.go:455 +0x8f0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x1400045e500, {0x1018d04c8, 0x140005bcb70}, 0x1400081e090, {0x1017c7800, 0x140005588b8})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:883 +0xe0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400045e500, {0x1018d04c8, 0x140005bcb70}, 0x140005181a0, 0x140005ae900, {0x1017c7800, 0x140005588b8})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/resource.go:990 +0x6bc
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400000e090, {0x1018d04c8?, 0x140005bc9c0?}, 0x1400057aeb0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/grpc_provider.go:1499 +0xc30
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x1018e1b40?, 0x1400000e090?}}, {0x1018d04c8, 0x140005bc9c0}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf5to6server/tf5to6server.go:39 +0x58
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x1400011ac60, {0x1018d04c8?, 0x140005bc6f0?}, 0x1400057ae60)
        github.com/hashicorp/terraform-plugin-mux@v0.21.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x178
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x14000468780, {0x1018d04c8?, 0x140003ebe30?}, 0x140002f3500)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:944 +0x274
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x101881740, 0x14000468780}, {0x1018d04c8, 0x140003ebe30}, 0x140002f3480, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:789 +0x1b8
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000236200, {0x1018d04c8, 0x140003ebda0}, 0x14000981bc0, 0x14000366750, 0x1024a59a8, 0x0)
        google.golang.org/grpc@v1.75.1/server.go:1431 +0xbd0
google.golang.org/grpc.(*Server).handleStream(0x14000236200, {0x1018d0d18, 0x14000313d40}, 0x14000981bc0)
        google.golang.org/grpc@v1.75.1/server.go:1842 +0x858
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.75.1/server.go:1061 +0x74
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 30
        google.golang.org/grpc@v1.75.1/server.go:1072 +0x120

Error: The terraform-provider-scaleway_v2.69.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

--- a/internal/services/applesilicon/server.go
+++ b/internal/services/applesilicon/server.go
@@ -452,7 +452,16 @@ func ResourceAppleSiliconServerUpdate(ctx context.Context, d *schema.ResourceDat
 	}
 
 	if d.HasChange("runner_ids") {
-		req.AppliedRunnerConfigurations.RunnerConfigurationIDs = d.Get("runner_ids").([]string)
+		if req.AppliedRunnerConfigurations == nil {
+			req.AppliedRunnerConfigurations = &applesilicon.AppliedRunnerConfigurations{}
+		}
+
+		runnerIDs := d.Get("runner_ids")
+		if runnerIDs != nil {
+			req.AppliedRunnerConfigurations.RunnerConfigurationIDs = locality.ExpandIDs(runnerIDs)
+		} else {
+			req.AppliedRunnerConfigurations.RunnerConfigurationIDs = []string{}
+		}
 	}
 
 	_, err = asAPI.UpdateServer(req, scw.WithContext(ctx))
@@ -494,6 +503,11 @@ func ResourceAppleSiliconServerDelete(ctx context.Context, d *schema.ResourceDat
 	}
 
 	err = detachAllPrivateNetworkFromServer(ctx, d, m, ID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, err = waitForAppleSiliconServer(ctx, asAPI, zone, ID, d.Timeout(schema.TimeoutDelete))
 	if err != nil {
 		return diag.FromErr(err)
 	}

--- a/internal/services/applesilicon/server_test.go
+++ b/internal/services/applesilicon/server_test.go
@@ -14,8 +14,10 @@ import (
 )
 
 var (
-	githubUrl   = os.Getenv("GITHUB_URL_AS")
-	githubToken = os.Getenv("GITHUB_TOKEN_AS")
+	githubUrl          = os.Getenv("GITHUB_URL_AS")
+	githubToken        = os.Getenv("GITHUB_TOKEN_AS")
+	githubUrlUpdated   = os.Getenv("GITHUB_URL_UPDATED_AS")
+	githubTokenUpdated = os.Getenv("GITHUB_TOKEN_UPDATED_AS")
 )
 
 func TestAccServer_Basic(t *testing.T) {
@@ -106,6 +108,46 @@ func TestAccServer_Runner(t *testing.T) {
 						runner_ids = [scaleway_apple_silicon_runner.main.id]
 					}
 				`, githubUrl, githubToken),
+				Check: resource.ComposeTestCheckFunc(
+					isServerPresent(tt, "scaleway_apple_silicon_server.main"),
+					resource.TestCheckResourceAttr("scaleway_apple_silicon_server.main", "name", "TestAccServerRunner"),
+					resource.TestCheckResourceAttr("scaleway_apple_silicon_server.main", "type", "M2-L"),
+					resource.TestCheckResourceAttr("scaleway_apple_silicon_server.main", "public_bandwidth", "1000000000"),
+					// Computed
+					resource.TestCheckResourceAttrSet("scaleway_apple_silicon_server.main", "ip"),
+					resource.TestCheckResourceAttrSet("scaleway_apple_silicon_server.main", "vnc_url"),
+					resource.TestCheckResourceAttrSet("scaleway_apple_silicon_server.main", "os_id"),
+					resource.TestCheckResourceAttrSet("scaleway_apple_silicon_server.main", "runner_ids.0"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
+					data "scaleway_apple_silicon_os" "devos" {
+						name = "devos-sequoia-15.6"
+					}
+
+					resource "scaleway_apple_silicon_runner" "main" {
+						name       = "TestAccRunnerGithub"
+						ci_provider   = "github"
+						url        = "%s"
+						token      = "%s"
+					}
+
+					resource "scaleway_apple_silicon_runner" "main-bis" {
+						name       = "TestAccRunnerGithub"
+						ci_provider   = "github"
+						url        = "%s"
+						token      = "%s"
+					}
+
+					resource scaleway_apple_silicon_server main {
+						name = "TestAccServerRunner"
+						type = "M2-L"
+						public_bandwidth = 1000000000
+						os_id = data.scaleway_apple_silicon_os.devos.id
+						runner_ids = [scaleway_apple_silicon_runner.main.id, scaleway_apple_silicon_runner.main-bis.id]
+					}
+				`, githubUrl, githubToken, githubUrlUpdated, githubTokenUpdated),
 				Check: resource.ComposeTestCheckFunc(
 					isServerPresent(tt, "scaleway_apple_silicon_server.main"),
 					resource.TestCheckResourceAttr("scaleway_apple_silicon_server.main", "name", "TestAccServerRunner"),

